primitives/trie/src/cache/shared_cache.rs

```diff
@@ -517,23 +616,73 @@ impl<H: Hasher> Clone for SharedTrieCache<H> {
 impl<H: Hasher> SharedTrieCache<H> {
 	/// Create a new [`SharedTrieCache`].
 	pub fn new(cache_size: CacheSize) -> Self {
-		let (node_cache_size, value_cache_size) = match cache_size {
-			CacheSize::Maximum(max) => {
-				// Allocate 20% for the value cache.
-				let value_cache_size_in_bytes = (max as f32 * 0.20) as usize;
-
-				(
-					CacheSize::Maximum(max - value_cache_size_in_bytes),
-					CacheSize::Maximum(value_cache_size_in_bytes),
-				)
-			},
-			CacheSize::Unlimited => (CacheSize::Unlimited, CacheSize::Unlimited),
-		};
+		let total_budget = cache_size.0;
+
+		// Split our memory budget between the two types of caches.
+		let value_cache_budget = (total_budget as f32 * 0.20) as usize; // 20% for the value cache
+		let node_cache_budget = total_budget - value_cache_budget; // 80% for the node cache
+
+		// Split our memory budget between what we'll be holding inline in the map,
+		// and what we'll be holding on the heap.
+		let mut value_cache_inline_budget = (value_cache_budget as f32 * 0.70) as usize;
```

Comment:
Is there any calculation for this? Like the size of a cached value vs the average heap size? Not sure if 30% for the map isn't too much?

Reply:
I picked this empirically too by just running a node, looking at the cache stats and picking a number that roughly maximized the space used.

It's obviously going to differ based on what's exactly cached, and it's probably not *the* optimal choice, but should be good enough for now. We can finetune it later.